Add basic support for Microchip PIC32CM MC devices

kinsamanka opened the pull request:
This PR adds basic UART and GPIO support for PIC32CM MC devices, a cost-optimized version of the SAM C2x family.

kinsamanka committed:
drivers: pinctrl_sam0: remove dependency on CONFIG_ASF

pinctrl_sam0.c requires atmel_sam_pinctrl.h to compile,
and this header is only available when CONFIG_ASF is enabled.

doc build fix applied:
    - unbalanced grouping commands
    - argument 'pin' of command @param is not found ...

Origin: hal_atmel
URL: https://github.com/zephyrproject-rtos/hal_atmel
commit: 2c45d61328a samv71b: patch: Fix SmcCs struct naming

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kinsamanka committed:
dts: add minimal support for PIC32CM MC family

Add minimal devicetree for PIC32CM MC devices,
a cost-optimized version of the SAM C2x family.

Supported peripherals:
    - flash
    - gpio
    - uart

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kinsamanka committed:
soc: add minimal support for PIC32CM MC family

These are the minimum code required to compile a basic firmware.

Supported peripherals:
    - flash
    - gpio
    - uart

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

gmarull reviewed (changes requested):
pls fix commit messages

```
dts: add minimal support for PIC32CM MC family
Origin: Original
```

nordicjm commented on soc/microchip/pic32cmmc/pic32cmmc00/soc.c:
```diff
@@ -0,0 +1,41 @@
+{
+	uint32_t data;
+
+	/* configure flash wait state to 3 @ 48MHz */
```
capitalise first letters of comments

nordicjm commented on soc/microchip/pic32cmmc/Kconfig.defconfig:
```diff
@@ -7,6 +7,9 @@ if SOC_FAMILY_MICROCHIP_PIC32CMMC
 
 rsource "*/Kconfig.defconfig"
 
+config GPIO
+	default y
+
```
why is this needed?

kinsamanka replied:
I just copied from sam0.

Is forced-pushing acceptable here to make the changes?

nordicjm replied:
Yes, rebase, make changes, force push

nordicjm commented on dts/arm/microchip/pic32cmmc.dtsi:
```diff
@@ -97,6 +97,50 @@
+			status = "disabled";
+		};
+
+		sercom1: sercom@42000800 {
```
why is no board added? None of this code is tested in CI

kinsamanka replied:
Do I need to include the board for this PR?

kinsamanka committed:
boards: initial support for PIC32CM CURIOSITY NANO

Add basic support for the PIC32CM MC Curiosity Nano Evaluation Kit to
run the blinky app.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

gmarull commented on soc/microchip/pic32cmmc/CMakeLists.txt:
```diff
@@ -1,4 +1,6 @@
 # Copyright (c) 2025 GP Orcullo
 # SPDX-License-Identifier: Apache-2.0
 
+zephyr_sources(../../atmel/sam0/common/soc_port.c)
```
create common directory for both vendors?

kinsamanka replied:
I'm not sure on how to handle this one as Atmel is owned by Microchip. I don't know how they will consolidate this.

kinsamanka committed:
drivers: pinctrl_sam0: refactor port driver

Removed dependency on the soc/atmel/sam0 directory to allow driver
reuse by other SOCs.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

nandojve commented:
Before this be merged I would like Nyias from MCHP be consulted.

kinsamanka committed:
drivers: intc_sam0_eic: remove dependency on ASF

Removing ASF dependency lets other SOCs reuse this driver.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kinsamanka committed:
drivers: flash_sam0: remove dependency on ASF

Removing ASF dependency lets other SOCs reuse this driver.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kinsamanka committed:
drivers: uart_sam0: remove dependency on ASF

Removing ASF dependency lets other SOCs reuse the driver.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kinsamanka committed:
drivers: gpio_sam0: remove dependency on ASF

Removing ASF dependency lets other SOCs reuse this driver.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kartben reviewed (changes requested):
The board being added will need a documentation page

kinsamanka committed:
drivers: i2c_sam0: remove dependency on ASF

Removing ASF dependency lets other SOCs reuse this driver.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kinsamanka committed:
boards: curiosity_nano_mc: add i2c and spi support

Enables I2C master and SPI master support.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>

kinsamanka committed:
drivers: spi_sam0: remove dependency on ASF

Removing ASF dependency lets other SOCs reuse this driver.

Signed-off-by: GP Orcullo <kinsamanka@gmail.com>
